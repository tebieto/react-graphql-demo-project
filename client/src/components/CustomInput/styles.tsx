import { TextField } from '@material-ui/core';
import styled from 'styled-components';

export const CustomInputContainer = styled(TextField)`
  fieldset {
    border-radius: 25px;
  }
`;
