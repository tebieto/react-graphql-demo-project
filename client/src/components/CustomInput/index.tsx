import React from 'react';
import { CustomInputContainer } from './styles';

interface CustomInputProps {
  onChange: { (e: React.ChangeEvent<HTMLInputElement>): void };
  placeholder?: string;
  type?: string;
  value?: string;
  size?: 'small' | 'medium' | undefined;
  onKeyDown?: { (e: React.KeyboardEvent<HTMLDivElement>): void };
}

const CustomInput = (props: CustomInputProps): JSX.Element => {
  return <CustomInputContainer variant="outlined" fullWidth {...props} />;
};

export default CustomInput;
