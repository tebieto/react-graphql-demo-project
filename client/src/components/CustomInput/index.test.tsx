import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CustomInput from '.';

it('renders correctly', () => {
  const onChange = () => null;
  const tree = shallow(<CustomInput value="Test" onChange={onChange} />);
  expect(toJson(tree)).toMatchSnapshot();
});
