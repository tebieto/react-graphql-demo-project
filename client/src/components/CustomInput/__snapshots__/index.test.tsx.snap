// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(WithStyles(ForwardRef(TextField)))
  InputProps={
    Object {
      "endAdornment": <WithStyles(ForwardRef(InputAdornment))
        position="end"
      >
        <Memo
          className="icon"
        />
      </WithStyles(ForwardRef(InputAdornment))>,
    }
  }
  fullWidth={true}
  onChange={[Function]}
  value="Test"
  variant="outlined"
/>
`;
